sites/kit.svelte.dev/src/lib/search/Search.svelte

```diff
@@ -0,0 +1,253 @@
+<script>
+	import { onMount } from 'svelte';
+	import flexsearch from 'flexsearch';
+
+	let searching = false;
+	let ul;
+
+	let query = '';
+	let results = [];
+	let selected = 0;
+
+	let index;
+	let lookup;
+
+	onMount(async () => {
+		const response = await fetch('/content.json');
+		const { blocks } = await response.json();
+
+		index = new flexsearch.Index({
+			tokenize: 'forward'
+		});
+
+		lookup = new Map();
+
+		for (const block of blocks) {
+			lookup.set(block.href, block);
+			index.add(block.href, block.content);
+		}
+	});
+
+	function update(e) {
+		query = e.target.value;
+		results = (index ? index.search(query) : []).map((href) => lookup.get(href));
+		selected = 0;
+	}
+
+	function escape(text) {
+		return text.replace(/</g, '&lt;').replace(/>/g, '&gt;');
+	}
+
+	function excerpt(content, query) {
+		const index = content.toLowerCase().indexOf(query.toLowerCase());
+		if (index === -1) {
+			return content.slice(0, 100);
+		}
+
+		const prefix = index > 20 ? `...${content.slice(index - 15, index)}` : content.slice(0, index);
+		const suffix = content.slice(
+			index + query.length,
+			index + query.length + (80 - (prefix.length + query.length))
+		);
+
+		return (
+			escape(prefix) +
+			`<mark>${escape(content.slice(index, index + query.length))}</mark>` +
+			escape(suffix)
+		);
+	}
+
+	function scroll_into_view() {
+		const li = ul && ul.children[selected];
+		if (li) {
+			const ul_rect = ul.getBoundingClientRect();
+			const li_rect = li.getBoundingClientRect();
+
+			const d_top = ul_rect.top - li_rect.top;
+			const d_bottom = ul_rect.bottom - li_rect.bottom;
+
+			if (d_top > 0) ul.scrollTop -= d_top;
+			if (d_bottom < 0) ul.scrollTop -= d_bottom;
+		}
+	}
+</script>
+
+<svelte:window
+	on:keydown={(e) => {
+		if (e.code === 'KeyK' && e.metaKey) {
+			e.preventDefault();
+			searching = !searching;
+		}
+	}}
+/>
+
+<div class="search-container">
+	<input
+		on:input={(e) => {
+			searching = true;
+			query = e.target.value;
+			e.target.value = '';
+		}}
+		on:click={() => (searching = true)}
+		type="search"
+		placeholder="Search (Cmd-K)"
+	/>
+</div>
+
+{#if searching}
+	<div class="modal-background" on:click={() => (searching = false)}>
+		<div class="search-box" on:click={(e) => e.stopPropagation()}>
+			<!-- svelte-ignore a11y-autofocus -->
+			<input
```

Comment:
This is going to need some a11y improvements + screen reader testing to make sure it behaves properly. Comboboxes like this are one of the trickier patterns from an a11y perspective. I'll do some research and get back to you.

Potential resources/patterns I'll be diving into:
- [WAI-ARIA combobox pattern](https://www.w3.org/TR/wai-aria-practices-1.1/#combobox)
- [React ARIA - Creating an accessible autocomplete experience](https://react-spectrum.adobe.com/blog/building-a-combobox.html)
- [Lion combobox component](https://lion-web.netlify.app/components/inputs/combobox/overview/)
- Other framework docs search widgets

Reply:
I changed my mind and don't think this is really a combo box. See my [new comment](https://github.com/sveltejs/kit/pull/3717#discussion_r800108915).